format/wasm/mappers.go

```diff
@@ -0,0 +1,211 @@
+package wasm
+
+import (
+	"errors"
+
+	"github.com/wader/fq/pkg/scalar"
+)
+
+var sectionIDToSym = &sectionIDToSymMapper{}
+
+type sectionIDToSymMapper struct {
+}
+
+func (m *sectionIDToSymMapper) MapScalar(s scalar.S) (scalar.S, error) {
+	v, ok := s.Actual.(uint64)
+	if !ok {
+		return s, errors.New("unexpected data type for section ID")
+	}
+
+	switch v {
+	case sectionIDCustom:
+		s.Sym = "custom section"
+	case sectionIDType:
+		s.Sym = "type section"
+	case sectionIDImport:
+		s.Sym = "import section"
+	case sectionIDFunction:
+		s.Sym = "function section"
+	case sectionIDTable:
+		s.Sym = "table section"
+	case sectionIDMemory:
+		s.Sym = "memory section"
+	case sectionIDGlobal:
+		s.Sym = "global section"
+	case sectionIDExport:
+		s.Sym = "export section"
+	case sectionIDStart:
+		s.Sym = "start section"
+	case sectionIDElement:
+		s.Sym = "element section"
+	case sectionIDCode:
+		s.Sym = "code section"
+	case sectionIDData:
+		s.Sym = "data section"
+	case sectionIDDataCount:
+		s.Sym = "data count section"
+	default:
+		s.Sym = "unknown section"
+	}
+
+	return s, nil
+}
+
+var valtypeToSymMapper = &valtypeToSym{}
+
+type valtypeToSym struct {
+}
+
+// valtype ::= t:numtype => t
+//          |  t:vectype => t
+//          |  t:reftype => t
```

Comment:
Maybe a go 1.19 godoc fmt thing?

Reply:
It is from WebAssembly's specification:

![image](https://user-images.githubusercontent.com/84613172/185734027-a63bcba5-efe1-4f07-96f1-5f0e614fe426.png)

Do you mean I should follow some godoc formatting here?

Reply:
Ah sorry it was about the linter complaining about godoc formatting, i like the comments with spec grammar 👍 to make the godoc formatter happy maybe they have to be verbatim somehow, maybe code "code blocks" see https://tip.golang.org/doc/comment? seem to work for me if i change:

```
// importdesc ::= 0x00 x:typeidx     => func x
//             |  0x01 tt:tabletype  => table tt
//             |  0x02 mt:memtype    => mem mt
//             |  0x03 gt:globaltype => global gt
``` 
to
```
// Some title
//
//  importdesc ::= 0x00 x:typeidx     => func x
//              |  0x01 tt:tabletype  => table tt
//              |  0x02 mt:memtype    => mem mt
//              |  0x03 gt:globaltype => global gt
```

Reply:
Got it, I'll update comments.